tests/test_agent.py

```diff
@@ -231,6 +232,7 @@ def test_calls_collect_logs_with_proper_mode(self, mock_log_collector, *args):
     def test_calls_collect_logs_on_valid_cgroups(self, mock_log_collector):
         try:
             CollectLogsHandler.enable_cgroups_validation()
+            mock_log_collector.run = Mock()
```

Comment:
run method has protocol initialization and that is internally making network calls which ls happening in loop due to retry setup on failures.